roles/puppet_agent/defaults/main.yml

```diff
@@ -0,0 +1,12 @@
+---
+puppet_package_name: puppet-agent
+puppet_package_state: installed
+puppet_package_manage: true
+
+puppet_group_name: puppet
+puppet_group_state: present
+puppet_group_manage: true
+
+puppet_bootstrap_manage: true
+puppet_bootstrap_server: ""
```

Comment:
You can leave this empty and do a `is not defined` check. That is how I typically see it in Ansible.

Reply:
I dislike that pattern because then I don't know what variables I can change. It's also how foreman_ansible discovers variables: read defaults.

Reply:
Right, what I meant was you can leave the value empty:

```
puppet_bootstrap_server:
```